### ISSUE TITLE
NVDA doesn't read skype messages correctly in other languages than english

### ISSUE THREAD

**Issue by francipvb** [nvaccess/nvda#5648]
- [ ] Change the Regular expression to be translatable
- [ ] Test if the RE can be translated
- [ ] Make translations for the languages.

Hi,

I have been taken a look at the source for the skype app module, and I found thisat the line 117.

``` python
    RE_MESSAGE = re.compile(r"^From (?P<from>.*), (?P<body>.*), sent on (?P<time>.*?)(?: Edited by .* at .*?)?(?: Not delivered|New)?$")
    def reportMessage(self, text):
        # Messages are ridiculously verbose.
        # Strip the time and other metadata if possible.
        m = self.RE_MESSAGE.match(text)
        if m:
            text = "%s, %s" % (m.group("from"), m.group("body"))
        ui.message(text)
```

I think that the RE needs to be translated to enable the correct speaking on all other languages. I can do it and test it, but I don't have the patiense to setup a development environment for NVDA.

Cheers.

**Comment by jcsteh** [comment]
We're aware of this issue. However, making the regular expression translatable is not the correct solution because:

* If a translator breaks the regular expression syntax (which is quite easy to do), it will break the code for anyone using that language. This is very bad.
* Translators aren't necessarily familiar with specific applications. They need to know the exact text Skype uses in their language; there can be more than one possible translation.
* If there isn't a version of Skype for a specific language but the translator tried to translate the regular expression anyway, it won't work.
* Ditto if the user uses one language for NVDA but chooses to use a different language for Skype for some reason.

The correct solution is to have the regular expression in the code support multiple languages. This can be done using the regexp `|` operator. This is annoying, since it means translators can't do this separately, but it prevents all of the above. It also won't work if Skype puts the pieces of text in a different order for different languages.

So, please provide the regular expression that works for your language. I will try to merge it with the English one.

**Comment by francipvb** [comment]
Hi,

OK, you're right, and I don't know anithing about gettext and the 
Regular Expressions. Another way is to provide a tuple of expressions 
with all of the regular expression inside it, and check them individually.

It's the RE for spanish version of Skype: `^De (?P<from>.*), 
(?P<body>.*), enviado el (?P<time>.*?)(?: Modificado por .* a las 
.*?)?(?: No entregado|Nuevo)?$`

As you can see, it's basically the same, but I don't know how it can 
change in other languages, for example Ja or DE or HE.

Cheers.
El 04/01/2016 a las 12:58 a.m., James Teh escribió:
> We're aware of this issue. However, making the regular expression
> translatable is not the correct solution because:
>
>   * If a translator breaks the regular expression syntax (which is quite
>     easy to do), it will break the code for anyone using that language.
>     This is very bad.
>   * Translators aren't necessarily familiar with specific applications.
>     They need to know the exact text Skype uses in their language; there
>     can be more than one possible translation.
>   * If there isn't a version of Skype for a specific language but the
>     translator tried to translate the regular expression anyway, it
>     won't work.
>   * Ditto if the user uses one language for NVDA but chooses to use a
>     different language for Skype for some reason.
>
> The correct solution is to have the regular expression in the code
> support multiple languages. This can be done using the regexp |||
> operator. This is annoying, since it means translators can't do this
> separately, but it prevents all of the above. It also won't work if
> Skype puts the pieces of text in a different order for different languages.
>
> So, please provide the regular expression that works for your language.
> I will try to merge it with the English one.
>

-- 
¡Cuando tus fuerzas terminan, las de Dios comienzan!

**Comment by francipvb** [comment]
Hello,

Just making this issue alive ...

What about reading the incoming messages as they come up and let the user change it through dictionary?

It is a nice solution I think.

Cheers,

**Comment by leonardder** [comment]
Closing this in favour of duplicate #4951, as the current ticket requests for making the regex translatable, which is a no go according to @JCSTeh in https://github.com/nvaccess/nvda/issues/5648#issuecomment-168575699
